### PR TITLE
Auto-registered ops: Add ops returning views to `_syms_returning_views`

### PR DESCRIPTION
**Pull request by kiya00**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

As a follow-up to https://github.com/Lightning-AI/lightning-thunder/issues/957 (https://github.com/Lightning-AI/lightning-thunder/issues/957#issuecomment-2293521400)

When auto-registration is added there's no in-place support yet. As mentioned in #957, the operators that return views can interfere with in-place ops. This PR adds the relative auto-registered symbols to `_syms_returning_views` and `_syms_returning_runtime_dependently_views` and let the in-place pass deal with it.

### CODE REVIEW

File: thunder/torch/__init__.py [hunk 1]
```diff
@@ -5636,7 +5637,13 @@ def check_overlap_ops():
 
 
 # ref: https://pytorch.org/docs/stable/tensor_view.html
-_syms_returning_runtime_dependently_views: set[Symbol] = {reshape, contiguous, to, flatten}
+_syms_returning_runtime_dependently_views: set[Symbol] = {
```
Comment (mruberry) [inline thunder/torch/__init__.py:5640]: `_syms_that_may_return_views` might be a clearer name

A comment explaining what these operations are would be helpful, as well as an explanation of how this set works

File: thunder/torch/default_torch_ops.py [hunk 2]
```diff
@@ -756,3 +756,28 @@
         torch.fft.rfftn,
     ],
 }
+
+_auto_registered_operators_returning_views = [
```
Comment (mruberry) [inline thunder/torch/default_torch_ops.py:760]: A comment here, like what this list does, what it means, and how it's generated, would be helpful

**Review by mruberry — COMMENTED** [review]
Hey @kiya00! I just made a couple suggestions for comments to improve readability

File: thunder/torch/default_torch_ops.py (hunk 2 above)
Comment (kiya00) [reply thunder/torch/default_torch_ops.py:764]: Hi @mruberry , the list of ops that return or may return tensor views are used in the functionalization, I pointed it to the in-place check function that uses this list

**Review by mruberry — APPROVED** [review]
Cool!

@IvanYashchuk, would you like to take a look?

**Pull request merged**